### PR TITLE
`OBJ_obj2txt()`: fix off-by-one documentation of the result - backport to 3.0 and 1.1.1

### PR DESCRIPTION
**Pull request by DDvO**
This backports the doc improvements of #17188 to 3.0 and 1.1.1.

### CODE REVIEW

File: doc/man3/OBJ_nid2obj.pod [hunk 1]
```diff
@@ -152,9 +154,10 @@ a NID or B<NID_undef> on error.
 
 OBJ_add_sigid() returns 1 on success or 0 on error.
 
-OBJ_obj2txt() returns the number of bytes written to I<buf> if I<buf_len> is big enough. 
-Otherwise, the result is truncated and the total amount of space required is returned. 
-It also returns -1 on error.
+i2t_ASN1_OBJECT() an OBJ_obj2txt() return -1 on error.
+On success, they return the lentgh of the string written to I<buf> if I<buf> is
```
Comment (t8m) [inline doc/man3/OBJ_nid2obj.pod:158]: Typo: length

Comment (DDvO) [reply doc/man3/OBJ_nid2obj.pod:158]: Oops, fixed.

**Review by t8m — APPROVED** [review]

**Review by paulidale — APPROVED** [review]

**Comment by t8m** [comment]
Merged to 3.0. This does not apply cleanly to 1.1.1 so that will require a separate PR. @DDvO would you please create one?

**Comment by DDvO** [comment]
> Merged to 3.0. 

Thanks.

>This does not apply cleanly to 1.1.1 so that will require a separate PR. @DDvO would you please create one?

Yep, done in #17221.

**Pull request closed without merging**